iroh-embed/src/lib.rs

```diff
@@ -0,0 +1,83 @@
+use anyhow::Result;
+use iroh_api::Api;
+use iroh_one::mem_p2p;
+use iroh_one::mem_store;
+use iroh_rpc_types::Addr;
+use tokio::task::JoinHandle;
+
+// TODO: This struct with all fields being pub is not suitable for an API as it is not
```

Comment:
isn’t there non_exhaustive on structs now?

Reply:
yes, though if you use that you can not create the struct outside of the crate that defines it.  So you'd have to switch to a builder of some sorts anyway.